cartesapp/cli.py

```diff
@@ -0,0 +1,466 @@
+import os
+import logging
+from typing import Optional, List, Annotated
+import traceback
+import typer
+from enum import Enum
+
+from .setting import SETTINGS_TEMPLATE
+from .manager import Manager
+
+LOGGER = logging.getLogger(__name__)
+
+###
+# AUX
+
+class NodeMode(str, Enum):
+    dev = "dev"
+    reader = "reader"
+    full = "full"
+    sunodo = "sunodo"
+
+SUNODO_LABEL_PREFIX = "io.sunodo"
+CARTESI_LABEL_PREFIX = "io.cartesi.rollups"
+
+UNITS = {"b": 1, "Kb": 2**10, "Mb": 2**20, "Gb": 2**30, "Tb": 2**40}
+
+MACHINE_CONFIGFILE = ".machine_config.json"
+
+def parse_size(size):
+    import re
+    m = re.compile('(\d+)\s*(\S+)').match(size)
+    number = m.group(1)
+    unit = m.group(2)
+    return int(float(number)*UNITS[unit])
+
+def build_image():
+    import subprocess, tempfile
+    with tempfile.NamedTemporaryFile(mode='w+') as idfile:
+        args = ["docker","build","--iidfile",idfile.name,"."]
+        result = subprocess.run(args)
+        if result.returncode > 0:
```

Comment:
Since negative values also represent that the process did not complete, I would use:
```suggestion
        if result.returncode != 0:
```
Or the [check_returncode()](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.check_returncode) method, which already raises an exception if code is non-zero.